[shashahchk] iP

shashahchk opened the pull request:
### DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro is your personal friendly assistant to help you manage tasks better!
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

**download** it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).

1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉

And it is _FREE_!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
added welcome message

Commit:
echo command back and exit when bye

shashahchk committed:
echo command back and exit when bye

shashahchk committed:
add, list

shashahchk committed:
mark as done

mark as done

shashahchk committed:
close scanner

shashahchk committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

shashahchk committed:
ToDos, Events, Deadlines

shashahchk committed:
add, list, made Task class

shashahchk committed:
fixed mark and unmark bug

shashahchk committed:
handle errors

shashahchk committed:
delete + auto test cases

shashahchk committed:
Automated text UI testing

shashahchk committed:
Merge remote-tracking branch 'origin/master'

shashahchk committed:
delete

shashahchk committed:
Merge remote-tracking branch 'origin/master'

shashahchk committed:
enums and debugged auto test cases

shashahchk committed:
added javadocs and debugged

shashahchk committed:
Save data into hard disc

shashahchk committed:
Merge branch 'branch-Level-7'
Load and save tasks data from and to hard disc

shashahchk committed:
Store date in LocalDateTime format

shashahchk committed:
implement oop

shashahchk committed:
add packages

shashahchk committed:
Merge remote-tracking branch 'origin/add-gradle-support'
merge and get gradle

shashahchk committed:
add junit test cases

shashahchk committed:
modify gradle to package JAR

shashahchk committed:
add javadoc

shashahchk committed:
tweak code to comply with coding standard

shashahchk committed:
add find functionality to find matching tasks

shashahchk committed:
resolve merge conflicts

shashahchk committed:
resolve merge conflicts

shashahchk committed:
add checkstyle

shashahchk committed:
add gui

NereusWB922 commented on src/main/java/Duke/tasks/Deadlines.java:
```diff
@@ -0,0 +1,89 @@
+ * A Deadlines object is a task with a deadline represented by a description and
+ * a date.
+ */
+public class Deadlines extends Task {
```
Perhaps having singular class name?

NereusWB922 commented on src/test/java/Duke/tasks/EventTest.java:
```diff
@@ -0,0 +1,29 @@
+     * @throws DukeException if there is an error during event creation.
+     */
+    @Test
+    public void eventCreation() throws DukeException {
```
Perhaps using test method naming convention to improve readability?

NereusWB922 commented on src/main/java/Duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    public Task parseTaskFromFile(String line) throws DukeException {
```
Good method naming, easy to understand the function of the method.

NereusWB922 commented on src/main/java/Duke/tasks/Task.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Toggles the completion status of the task.
+     */
+    public void toggleCompleted() {
```
Perhaps using an easy pronounce method name like toggleStatus?

NereusWB922 reviewed (commented):
Looks good!

amosting commented on src/main/java/Duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+import Duke.tasks.TaskList;
+public class Parser {
+    public CommandType getCommandType(String userCommand) {
+        if ("bye".equals(userCommand)) {
```
Can consider using switch-case instead for cleaner-looking code. If-else with the curly braces and indented could start to look a bit clunky.

amosting commented on src/main/java/Duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        Task task;
+    
+        switch (commandType) {
```
good use of switch case!

amosting commented on src/main/java/Duke/tasks/TaskManager.java:
```diff
@@ -0,0 +1,144 @@
+package Duke.tasks;
```
Good use of packages to organise the different classes!

amosting commented on src/main/java/core/Duke.java:
```diff
@@ -0,0 +1,107 @@
+    private TaskManager tm;
+
+
+    public enum CommandType {
```
good use of enum!

shashahchk committed:
Add extra features in GUI

The interface is lacking in visual information. It is difficult to keep
track of the conversation given the similar looking texts.

Dividing the interaction into the user's input and the program's output
makes the communication more clear and convenient.

To make the interaction more pleasing, let's add graphical user
interface that simulates communicating with another person.

shashahchk committed:
Add assertions to check validity in the program

There may be errors in the code that are hard to trace.

Adding assertions along the program help to catch errors faster and
allow easier debugging.

Let's add assertions to check for validity of input.

shashahchk committed:
Improve code quality

The code has certain unused methods and bad naming.

To make the code more comprehensive and easier to follow, method and
variable names should be improved.

Let's remove unused methods and change naming to fit coding standard.

shashahchk committed:
Merge pull request #2 from shashahchk/branch-A-Assertions

Add assertions to check validity in the program

shashahchk committed:
Merge pull request #3 from shashahchk/branch-A-CodeQuality

Improve code quality

shashahchk committed:
Remove unnecessary code

shashahchk committed:
Merge pull request #4 from shashahchk/Merging-PRs

Remove unnecessary code

shashahchk committed:
Improve gui

shashahchk committed:
Add archive function

shashahchk committed:
improve gui images and background

shashahchk committed:
fix long methods and remove unused methods

shashahchk committed:
change bg color gui

shashahchk committed:
add test cases for parser and deadline, event, todo tasks

shashahchk committed:
add user guide

shashahchk committed:
improve user guide

shashahchk committed:
improve user guide and add ui into docs

shashahchk committed:
remove duplicate files

shashahchk committed:
fix adjustable dimensions gui and package naming

shashahchk committed:
improve gui

shashahchk committed:
fix styling

shashahchk committed:
add more testing

shashahchk committed:
edit readme
